piksi_tools/console/tracking_view.py

```diff
@@ -133,104 +133,123 @@ class TrackingView(CodeFiltered):
                 Item('legend_visible', label="Show Legend:"),
                 CodeFiltered.get_filter_group(), )))
 
+    def clean_cn0(self, t):
+        assert self.CN0_lock.locked()
+        for k in list(self.CN0_dict.keys()):
```

Comment:
`list()` shouldn't be necessary here, `.keys()` returns a copy
https://docs.python.org/2.7/library/stdtypes.html#dict.keys

Reply:
It's necessary here since CN0_dict is being modified inside the iterator

Reply:
ah, my bad; i thought this was python 2

Reply:
Well, it is and it isn't, I think this backward (and forward) compatible with both.  Going forward we're going to want Python 3 support, so I think it's safe to introduce some compat stuff now.